source/fundamentals/builders/indexes.txt

```diff
@@ -11,3 +11,181 @@ Indexes Builders
+   :language: java
+   :dedent:
+   :start-after: begin ascendingIndex
+   :end-before: end ascendingIndex
+
+.. _descending-indexes:
+
+Descending Indexes
+------------------
+
+A descending index enables you to sort query results by the value of the
+indexed field(s) from largest to smallest.
+
+In order to create a descending index, first call the
+:java-core-api:`descending() <com/mongodb/client/model/Indexes.html#descending(java.lang.String...)>`
+builder method to create a ``Bson`` instance that represents the index
+document, passing the name or names of the fields you want to index.
+Then, call the ``createIndex()`` method on the collection, passing the ``Bson``
+instance that contains the index document.
+
+The following example specifies a descending index on the ``capacity`` field: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/Indexes.java
+   :language: java
+   :dedent:
+   :start-after: begin descendingIndex
+   :end-before: end descendingIndex
+
+.. _index-compound-indexes:
+
+Compound Indexes
+----------------
+
+In order to create a compound index, first call the
+:java-core-api:`compoundIndex() <com/mongodb/client/model/Indexes.html#compoundIndex(org.bson.conversions.Bson...)>`
+builder method to create a ``Bson`` instance that represents the index
+document, passing the names of the fields you want to index. Then, call
+the ``createIndex()`` method on the collection, passing the ``Bson``
+instance that contains the index document. 
+
+The following example specifies a compound index composed of
+descending index on the ``capacity`` and ``year`` field, followed
+by an ascending index on the ``name`` field: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/Indexes.java
+   :language: java
+   :dedent:
+   :start-after: begin compoundIndexExample
+   :end-before: end compoundIndexExample
+
+.. _index-text-indexes:
+
+Text Indexes
+------------
+
+A text index groups documents by the text in the indexed field.
+
+In order to create a text index, first call the
+:java-core-api:`text() <com/mongodb/client/model/Indexes.html#text(java.lang.String)>`
+builder method to create a ``Bson`` instance that represents the index
+document, passing the name of the fields you want to index. Then, call
+the ``createIndex()`` method on the collection, passing the ``Bson``
+instance that contains the index document. 
+
+The following example specifies a text index key on the "theaters" field:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/Indexes.java
+   :language: java
+   :dedent:
+   :start-after: begin textIndex
+   :end-before: end textIndex
+
+.. _index-hashed-indexes:
+
+Hashed Indexes
+--------------
+
+A hashed index groups documents by the hash value in the indexed field.
+
+In order to create a hashed index, first call the
+:java-core-api:`hashed() <com/mongodb/client/model/Indexes.html#hashed(java.lang.String)>`
```

Comment:
Whoops, looks like the link to the MongoDB docs contained in the Java API docs is broken (https://docs.mongodb.com/manual/core/hashed/).

Any way we can get that link to redirect to https://docs.mongodb.com/manual/core/index-hashed/index.html?

Reply:
The link I have there is actually: https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Indexes.html#hashed(java.lang.String) 

But if you'd like, I can switch it to the link you requested

Reply:
Sorry, I wasn't clear.  It's the link from the driver API docs to the MongoDB docs that is broken.  I was hoping that we can make that link work again as we can't easily update the Javadoc for already-released versions of the driver.

Reply:
I created a ticket to get a redirect created: https://jira.mongodb.org/browse/DOP-2067?filter=-2